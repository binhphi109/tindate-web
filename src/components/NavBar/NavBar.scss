.navbar {
  padding: 8px;
  display: flex;

  .navbarButton {
    display: flex;
    justify-content: center;
    flex: 1;

    i {
      width: 24px;
    }
  }
}
