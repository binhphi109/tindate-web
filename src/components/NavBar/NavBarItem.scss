.navbarItem {
  flex: 1;
  display: flex;
  align-items: center;
  justify-content: center;

  i {
    width: 24px;
  }

  .navbarTitle {
    font-weight: bold;
    color: #fe5858;
  }
}
