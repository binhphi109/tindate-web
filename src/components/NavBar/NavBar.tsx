import React from "react";
import clsx from "clsx";
import "./NavBar.scss";

type Props = {
  className?: string;
  children: any;
};

function NavBar({ className, children }: Props) {
  return <div className={clsx("navbar", className)}>{children}</div>;
}

NavBar.defaultProps = {
  className: null,
};

export default NavBar;
