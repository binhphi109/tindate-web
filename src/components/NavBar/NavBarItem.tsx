import React, { MouseEventHandler } from "react";
import { IconButton } from "@material-ui/core";
import clsx from "clsx";
import "./NavBarItem.scss";

type Props = {
  className?: string;
  icon?: string;
  iconColor?: string;
  title?: string;
  color?: "inherit" | "primary" | "secondary" | "default";
  onClick?: MouseEventHandler;
};

function NavBarItem({ className, icon, iconColor, title, color, onClick }: Props) {
  return (
    <div className={clsx("navbarItem", className)} onClick={onClick}>
      {icon && (
        <IconButton size="medium" color={color}>
          <i className={icon} style={{ color: iconColor }} />
        </IconButton>
      )}
      {title && <div className="navbarTitle">{title}</div>}
    </div>
  );
}

NavBarItem.defaultProps = {
  className: null,
  icon: null,
  iconColor: null,
  title: null,
  color: "default",
  onClick: null,
};

export default NavBarItem;
