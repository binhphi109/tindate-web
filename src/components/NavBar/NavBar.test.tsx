import React from "react";
import { shallow } from "enzyme";
import NavBar from "./NavBar";
import NavBarItem from "./NavBarItem";

describe("components/navbar/NavBar", () => {
  test("should match snapshot with icon", () => {
    const onFirstClick = jest.fn();
    const onThirdClick = jest.fn();
    const wrapper = shallow(
      <NavBar>
        <NavBarItem icon="fas fa-tasks" onClick={onFirstClick} />
        <NavBarItem icon="fas fa-fire" color="primary" />
        <NavBarItem icon="far fa-comment" onClick={onThirdClick} />
      </NavBar>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("should match snapshot with title", () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <NavBar>
        <NavBarItem icon="fas fa-chevron-left" onClick={onClick} />
        <NavBarItem title="Likes" />
        <NavBarItem />
      </NavBar>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
