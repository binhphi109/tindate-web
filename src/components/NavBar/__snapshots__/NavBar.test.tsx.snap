// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/navbar/NavBar should match snapshot with icon 1`] = `
<div
  className="navbar"
>
  <NavBarItem
    className={null}
    color="default"
    icon="fas fa-tasks"
    iconColor={null}
    onClick={[MockFunction]}
    title={null}
  />
  <NavBarItem
    className={null}
    color="primary"
    icon="fas fa-fire"
    iconColor={null}
    onClick={null}
    title={null}
  />
  <NavBarItem
    className={null}
    color="default"
    icon="far fa-comment"
    iconColor={null}
    onClick={[MockFunction]}
    title={null}
  />
</div>
`;

exports[`components/navbar/NavBar should match snapshot with title 1`] = `
<div
  className="navbar"
>
  <NavBarItem
    className={null}
    color="default"
    icon="fas fa-chevron-left"
    iconColor={null}
    onClick={[MockFunction]}
    title={null}
  />
  <NavBarItem
    className={null}
    color="default"
    icon={null}
    iconColor={null}
    onClick={null}
    title="Likes"
  />
  <NavBarItem
    className={null}
    color="default"
    icon={null}
    iconColor={null}
    onClick={null}
    title={null}
  />
</div>
`;
